File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.java
class Solution {
    public int maxSum(int[] nums) {
        //given an array select all unique numbers and return their sum, 
        // if all negative just return the maximum number 

        int max = Integer.MIN_VALUE;
        int n = nums.length;

        for(int i = 0; i < n; i += 1){
            max = Math.max(nums[i], max);
        }

        if(max <= 0){
            return max;
        }
        int sum = 0;
        HashSet<Integer> set = new HashSet<>();


        for(var v : nums){
            if(v > 0 && !set.contains(v)){
                sum += v;
                set.add(v);
            }   
        }

        return sum;
    }
}

File: 3788-maximum-unique-subarray-sum-after-deletion/3788-maximum-unique-subarray-sum-after-deletion.cpp
class Solution {
public:
    int maxSum(vector<int>& nums) {
        unordered_set<int>set;
        int cnt =0, mini = INT_MAX;
        for(auto val : nums){
            if(val < 0){
                cnt++;
            }else{
                set.insert(val);
            }
            mini = max(mini, val);
        }
        if(cnt == nums.size())
            return mini;
        //it is always better to consider the while array
        //delete all negatives,
        //delete all duplicates
        int sum = 0;
        for(auto val : set)
            sum += val;
        
        return sum;
    }
};